### ISSUE TITLE
[Bug]: Connections established to 1e100.net & googleusercontent.com

### ISSUE THREAD

**Issue by jnolan7599** [FreeTubeApp/FreeTube#3015]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

Hello,

I'm seeing ESTABLISHED connections to 1e100.net, googleusercontent.com when using a Invidious instance as the API. I do not see these connections when using said instance's web site. This was identified using System Explorer 7.0.0.x.

(I seem to be unable to upload a jpg of such an occurrence.)

Do I have something misconfigured?

### Expected Behavior

I expected all connections to be proxied through my preferred Invidious instance per FreeTube's API.

### Issue Labels

API issue, inconsistent behavior

### FreeTube Version

v0.18.0 Beta (PortableApps)

### Operating System Version

Win 10 21H2

### Installation Method

PortableApps

### Primary API used

Invidious API

### Last Known Working FreeTube Version (If Any)

N/A

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by ChunkyProgrammer** [comment]
Is this what you see when watching videos? You'll need to enable `Proxy Videos Through Invidious` under Player Settings if that's the case.

**Comment by jnolan7599** [comment]
Thanks for the response.

_Fallback to Non-Preferred Backend on Failure_ was, fortunately, already disabled. Enabling _Proxy Videos Through Invidious,_ which, I might add may be better served by moving it into _General Settings_, doesn't seem to make a difference even after closing out FreeTube & restarting. I'm still seeing a connection to _1.80.190.35.bc.googleusercontent.com_.

Is there anything else I'm missing?

**Comment by jnolan7599** [comment]
Update: I flushed the DNS cache on my Windows box & router, three times each, sequentially. I'm still seeing two connection to _1.80.190.35.bc.googleusercontent.com_ when I then loaded FreeTube & tried playing a music video from my history page/db.

**Comment by ChunkyProgrammer** [comment]
Do you only notice this when playing a video?

**Comment by jnolan7599** [comment]
Correct. My DNS config automatically reboots at midnight; I've observed the same established connection within seconds of playing the first video after the first boot of my Windows system today.

This was not even 5 minutes ago. 

I'm wondering if this isn't somehow related to the 'title image' that's presented before hitting play. I don't see these kind of connections when looking over my history/subscription pages.

Replaying the same video in question again establishes a new connection to the aforementioned.

**Comment by jnolan7599** [comment]
Update: This is curious. I came [across a thread on privacytools IO reddit](https://libreddit.tiekoetter.com/r/privacytoolsIO/comments/jwglv1/blocking_access_to_all_sites/) fr 2020 which seems to imply the only way to block _bc.googleusercontent.com_ at that time was via a custom rule in Ublock Origin:

> The Ublock Origin * googleusercontent.com * block actually did the trick!

**Comment by absidue** [comment]
@jnolan7599 could you please check if this issue is fixed on the nightly builds? I'm not sure if pull request #3019 will have fixed this issue too.
You mentioned that you are using the PortableApps version of FreeTube, which we don't maintain, I don't know if that maintainer provides nightly builds. Our official nightly builds are here https://github.com/FreeTubeApp/FreeTube/actions/workflows/build.yml and here is the link to the latest one: https://github.com/FreeTubeApp/FreeTube/actions/runs/3822572021

**Comment by jnolan7599** [comment]
@absidue Yeah, it looks like @rddim 's v0.18.0 Dev Test 1 is from 2022 Nov 23.

I, uh, don't know how to pull or compile. There's so much rust on me I can barely handle Docker & navigate GitHub. If you could toss me a link to a msi/exe, I'd be happy give it a go.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> ... here is the link to the latest one: https://github.com/FreeTubeApp/FreeTube/actions/runs/3822572021

@jnolan7599 go to this link and scroll down the  page u'll find the installers there.

**Comment by jnolan7599** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc @ChunkyProgrammer Right, so I don't know if it's because I've not finished my first coffee of the day or I really am this dumb: I loaded _[runs/3822572021](https://github.com/FreeTubeApp/FreeTube/actions/runs/3822572021)._ I see the _Artifacts_ header. I see a list of _names, sizes_. I see the one I want to test, _freetube-0.18.0-nightly-2493-setup-x64.exe (60.8 MB)_. 

I do not see how I can download it. There's no link apparent. I presume they should be href'd as found on _Releases_ or similar as found in this very response. I checked in two different browsers, including Incognito mode to bypass all JS blocking/'shields'.

I'm pretty sure I'm encountering an ID10T error but your assistance really is appreciated.

**Comment by absidue** [comment]
Make sure you are logged in when trying to download the artifacts, GitHub only allows you to download them while logged in.

**Comment by jnolan7599** [comment]
_freetube-0.18.0-nightly-2493-setup-x64.exe_ (v0.18.0-nightly-2493 Beta) connects to the Google DNS, _1.80.190.35.bc.googleusercontent.com,_ even after confirming the instance set, _Fallback to Non-Preferred Backend on Failure_ disabled then _Player Settings,_ _Proxy Videos Through Invidious_ enabled.

It still occurs within seconds of 'pulling up' a video already visited &/or an unwatched video from  'trending' (eg: watch?v=iowiDhZJ_Jg).

I did **not** observe this connection after disabling _General Settings,_ _Enable Search Suggestions._

Should I file a new feature/bug report regarding consolidation of privacy settings/potential traffic leaking?

**Comment by jnolan7599** [comment]
A thought: I seem to recall though I have no source to reference ATM, that _googleusercontent.com_ hosts static content. Given I'm not seeing such connections when directly using a Invidious web site, how/where does FreeTube pull such YT-related assets such as channel icons/'avatars'?

**Comment by rddim** [comment]
@jnolan7599, @absidue 
I don't provide nightly builds but only the current pre/latest releases. If you want to use the latest nightly as portable (at your own risk) just download the latest *.7z build and replace the whole content in `FreeTubePortable\App\FreeTube`. Of course there is no guarantee that the latest will work as portable.

@jnolan7599 
>how/where does FreeTube pull such YT-related assets such as channel icons/'avatars'?

Try with `yt3.ggpht.com`. If you open your `profiles.db` file with some text editor you will see many links.

**Comment by jnolan7599** [comment]
@rddim Thanks for releasing the portable edition. This issue also applies to the installer builds too as noted.

I blocked _ggpht.com_ at the DNS level but am still seeing connections to _googleusercontent.com_. It seems assets are still being loaded though.

This was after the first boot of  the day today.

Am I missing anything else?

**Comment by ChunkyProgrammer** [comment]
> @rddim Thanks for releasing the portable edition. This issue also applies to the installer builds too as noted.
> 
> I blocked _ggpht.com_ at the DNS level but am still seeing connections to _googleusercontent.com_. It seems assets are still being loaded though.
> 
> This was after the first boot of the day today.
> 
> Am I missing anything else?

Could you try downloading a build from here?https://github.com/FreeTubeApp/FreeTube/actions/runs/3899899877
Some changes were made to invidious that might fix the issue

**Comment by jnolan7599** [comment]
@ChunkyProgrammer 

I still see connections after installing _freetube-0.18.0-nightly-2529-setup-x64.exe_ from the aforementioned run. DNS was flushed three times beforehand.
